backend/internal/controllers/book.go

```diff
@@ -285,3 +286,75 @@ func (c *BookController) GetBookClicks(rw http.ResponseWriter, req *http.Request
 	writeResponse(rw, http.StatusOK, bookAnalytics)
 
 }
+
+// fetches favorite books for the currently authenticated user (/books/favorite)
+func (c *BookController) GetFavorites(rw http.ResponseWriter, req *http.Request) {
+
+	userID, ok := req.Context().Value("user").(string)
+
+	if !ok {
+		writeResponse(rw, http.StatusInternalServerError, "error")
+		return
+	}
+
+	favorites, err := c.Books.FetchFavorites(req.Context(), userID)
+
+	if err != nil {
+		writeResponse(rw, http.StatusInternalServerError, "error")
+		return
+	}
+
+	writeResponse(rw, http.StatusOK, favorites)
+
+}
+
+// creates a new entry in the favorite books table (/books/favorites/{id})
+func (c *BookController) AddToFavorites(rw http.ResponseWriter, req *http.Request) {
+
+	userID, ok := req.Context().Value("user").(string)
+	var bookID string = chi.URLParam(req, "id")
```

Comment:
Can we validate the book ID? Right now this will 500 from the foreign key constraint but ideally we should return a 400 instead. You can check the error returned by InsertFavoriteBook or just try to query for the book with that ID before ever attempting an insert.